commit 2a5da9d00555beef8e5f6307cfcbfc029d45491e
Author: mundur <nmondoo97@gmail.com>
Date:   Wed Jul 8 22:38:10 2026 +0800

    Fixed #37203 -- Used normalized field names for composite pks in inspectdb.

diff --git a/tests/inspectdb/models.py b/tests/inspectdb/models.py
index 489704879f..a4838b508a 100644
--- a/tests/inspectdb/models.py
+++ b/tests/inspectdb/models.py
@@ -172,6 +172,12 @@ class CompositePKModel(models.Model):
     column_2 = models.IntegerField()
 
 
+class CompositePKModel2(models.Model):
+    pk = models.CompositePrimaryKey("column_1", "column_2")
+    column_1 = models.IntegerField(db_column="column-1")
+    column_2 = models.IntegerField(db_column="column-2")
+
+
 class DbOnDeleteModel(models.Model):
     fk_do_nothing = models.ForeignKey(UniqueTogether, on_delete=models.DO_NOTHING)
     fk_db_cascade = models.ForeignKey(ColumnTypes, on_delete=models.DB_CASCADE)
diff --git a/tests/inspectdb/tests.py b/tests/inspectdb/tests.py
index a0b0e54039..b3319dc4eb 100644
--- a/tests/inspectdb/tests.py
+++ b/tests/inspectdb/tests.py
@@ -722,6 +722,24 @@ class InspectDBTransactionalTests(TransactionTestCase):
         self.assertIn(f"column_1 = models.{field_type}()", output)
         self.assertIn(f"column_2 = models.{field_type}()", output)
 
+    def test_composite_primary_key_uses_normalized_column_names(self):
+        out = StringIO()
+        field_type = connection.features.introspected_field_types["IntegerField"]
+        call_command("inspectdb", "inspectdb_compositepkmodel2", stdout=out)
+        output = out.getvalue()
+        self.assertIn(
+            "pk = models.CompositePrimaryKey('column_1', 'column_2')",
+            output,
+        )
+        self.assertIn(
+            f"column_1 = models.{field_type}(db_column='column-1')",
+            output,
+        )
+        self.assertIn(
+            f"column_2 = models.{field_type}(db_column='column-2')",
+            output,
+        )
+
     def test_composite_primary_key_not_unique_together(self):
         out = StringIO()
         call_command("inspectdb", "inspectdb_compositepkmodel", stdout=out)
